rust_icu_uloc/src/lib.rs

```diff
@@ -763,4 +888,56 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_base_name() -> Result<(), Error> {
+        assert_eq!(
+            ULoc::try_from("en-u-tz-uslax-x-foo")?.base_name(),
+            ULoc::try_from("en")?
+        );
+        Ok(())
+    }
+
+    #[test]
+    fn test_uloc_mut() -> Result<(), Error> {
+        let loc = ULoc::for_language_tag("en-t-it-u-tz-uslax-x-foo")?;
+        let loc_mut = ULocMut::from(loc);
+        let loc = ULoc::from(loc_mut);
+        assert_eq!(ULoc::for_language_tag("en-t-it-u-tz-uslax-x-foo")?, loc);
+        Ok(())
+    }
+
+    #[test]
+    fn test_uloc_mut_changes() -> Result<(), Error> {
+        let loc = ULoc::for_language_tag("en-t-it-u-tz-uslax-x-foo")?;
+        let mut loc_mut = ULocMut::from(loc);
+        loc_mut.remove_unicode_keyvalue("tz");
+        let loc = ULoc::from(loc_mut);
+        assert_eq!(ULoc::for_language_tag("en-t-it-x-foo")?, loc);
+
+        let loc = ULoc::for_language_tag("en-u-tz-uslax")?;
+        let mut loc_mut = ULocMut::from(loc);
+        loc_mut.remove_unicode_keyvalue("tz");
+        let loc = ULoc::from(loc_mut);
+        assert_eq!(ULoc::for_language_tag("en")?, loc);
+        Ok(())
+    }
+
+    #[test]
+    fn test_uloc_mut_overrides() -> Result<(), Error> {
+        let loc = ULoc::for_language_tag("en-t-it-u-tz-uslax-x-foo")?;
+        let mut loc_mut = ULocMut::from(loc);
+        loc_mut.set_unicode_keyvalue("tz", "usnyc");
+        let loc = ULoc::from(loc_mut);
+        assert_eq!(ULoc::for_language_tag("en-t-it-u-tz-usnyc-x-foo")?, loc);
+
+        let loc = ULoc::for_language_tag("en-t-it-u-tz-uslax-x-foo")?;
+        let mut loc_mut = ULocMut::from(loc);
+        loc_mut.set_unicode_keyvalue("tz", "usnyc");
+        loc_mut.set_unicode_keyvalue("nu", "arabic");
+        let loc = ULoc::from(loc_mut);
+        assert_eq!(ULoc::for_language_tag("en-t-it-u-nu-arabic-tz-usnyc-x-foo")?, loc);
+        assert_eq!(ULoc::for_language_tag("en-t-it-u-tz-usnyc-nu-arabic-x-foo")?, loc);
+        Ok(())
+    }
```

Comment:
One more test case for completeness: calling `set_unicode_keyvalue()` on a locale that originally only has non-`u` extensions.